plugins/kotlin/j2k/k1.new/src/org/jetbrains/kotlin/nj2k/conversions/IsArrayOfCallConversion.kt

```diff
@@ -0,0 +1,47 @@
+// Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package org.jetbrains.kotlin.nj2k.conversions
+
+import org.jetbrains.kotlin.j2k.Nullability
+import org.jetbrains.kotlin.nj2k.NewJ2kConverterContext
+import org.jetbrains.kotlin.nj2k.RecursiveApplicableConversionBase
+import org.jetbrains.kotlin.nj2k.identifier
+import org.jetbrains.kotlin.nj2k.symbols.JKFieldSymbol
+import org.jetbrains.kotlin.nj2k.tree.*
+import org.jetbrains.kotlin.nj2k.types.*
+import org.jetbrains.kotlin.nj2k.unboxFieldReference
+
+internal class IsArrayOfCallConversion(override val context: NewJ2kConverterContext) :
+    RecursiveApplicableConversionBase(context) {
+
+    override fun applyToElement(element: JKTreeElement): JKTreeElement {
+        if (element !is JKIsExpression) return recurse(element)
+        if (!element.type.type.isArrayType()) return recurse(element)
+        val operand = element.expression
+        val operandType = operand.calculateType(typeFactory) ?: return recurse(element)
+        val type = element::type.detached().type
+
+        val isArrayOfExpression = JKQualifiedExpression(
+            operand.detached(element), JKCallExpressionImpl(
+                symbolProvider.provideMethodSymbol("kotlin.isArrayOf"),
+                JKArgumentList(), JKTypeArgumentList(type.arrayInnerType()!!)
+            )
+        )
+        val replacement = if (operandType.isArrayType()) {
+            isArrayOfExpression
+        } else {
+            JKBinaryExpression(
+                JKIsExpression(
+                    JKFieldAccessExpression(operand.identifier!! as JKFieldSymbol),
+                    JKClassType(
+                        symbolProvider.provideClassSymbol("kotlin.Array"),
+                        listOf(JKStarProjectionTypeImpl),
+                        Nullability.NotNull
+                    ).asTypeElement()
+                ),
+                isArrayOfExpression,
+                JKKtOperatorImpl(JKOperatorToken.ANDAND, typeFactory.types.boolean)
+            )
+        }
+        return recurse(replacement)
+    }
+}
```

Comment:
I don't know if the test differences in nullability are significant or kind of accidental. I think it makes sense to try to replicate the behavior of the original quick fix with regards to nullability (the `== true` part) and negation. And then hope that the future J2K nullability refactoring will provide the proper nullability of all types for this conversion.

So, I propose that the conversion should assume that types already have the correct nullability and properly handle both cases: nullable and not-null.

See the original commit with inspection and all relevant tests: https://github.com/JetBrains/intellij-community/commit/d1f935490a3d3bf3295ec75d0c06d0d8ffefe725